src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,321 @@
+import java.util.regex.Pattern;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtils;
+
+/**
+ * This class is used internally in the build process to write a property file
+ * with short descriptions (the first sentences) of TokenTypes constants.
+ * Request: 724871
+ * For IDE plugins (like the eclipse plugin) it would be useful to have
+ * a programmatic access to the first sentence of the TokenType constants,
+ * so they can use them in their configuration gui.
+ * @author Pavel Bludov
+ */
+public final class JavadocPropertiesGenerator {
+
+    /**
+     * The command line option to specify the output file.
+     */
+    private static final String OPTION_DEST_FILE = "destfile";
+
+    /**
+     * The width of the CLI help option.
+     */
+    private static final int HELP_WIDTH = 100;
+
+    /**
+     * This regexp is used to extract the first sentence from the text.
+     */
+    private static final Pattern END_OF_SENTENCE_PATTERN = Pattern.compile("(.*?[.?!])(\\s|$)");
+
+    /**
+     * Don't create instance of this class, use the {@link #main(String[])} method instead.
+     */
+    private JavadocPropertiesGenerator() {
+    }
+
+    /**
+     * TokenTypes.properties generator entry point.
+     * @param args the command line arguments
+     * @throws CheckstyleException if parser or lexer failed or if there is an IO problem
+     * @throws ParseException if the command line can not be passed
+     **/
+    public static void main(String... args)
+            throws CheckstyleException, ParseException {
+        final CommandLine commandLine = parseCli(args);
+        if (commandLine.getArgList().size() == 1) {
+            writePropertiesFile(commandLine.getArgList().get(0),
+                commandLine.getOptionValue(OPTION_DEST_FILE));
+        }
+        else {
+            printUsage();
+        }
+    }
+
+    /**
+     * Creates the .properties file from a .java file.
+     * @param inputFile .java file
+     * @param outputFile .properties file
+     * @throws CheckstyleException if a javadoc comment can not be parsed
+     */
+    private static void writePropertiesFile(String inputFile, String outputFile)
+            throws CheckstyleException {
+        PrintWriter writer = null;
+        try {
+            final DetailAST root = JavaParser.parseFile(new File(inputFile),
```

Comment:
root should be close to FOR.
All output/writer stuff should be separated by empty line from logic of iteration.

Reply:
done